novelist.py

```diff
@@ -0,0 +1,309 @@
+                    and eq.output.type==Identifier.ISBN)]
+
+        if not lookup_metadata:
+            self.log.error("Identifiers without an ISBN equivalent can't \
+                    be looked up with NoveList: %r" % identifier)
+            return None
+
+        return [metadata for metadata in lookup_metadata if metadata]
+
+    def lookup(self, identifier):
+        """Requests NoveList metadata for a particular identifier
+
+        :return: None, a Metadata object, or a list of Metadata objects
+        """
+
+        client_identifier = identifier.urn
+        if identifier.type != Identifier.ISBN:
+            return self.lookup_equivalent_isbns(identifier)
+
+        params = dict(
+            ClientIdentifier=client_identifier, ISBN=identifier.identifier,
+            version=self.version, profile=self.profile, password=self.password
+        )
+        url = self._build_query(params)
+        self.log.debug("NoveList lookup: %s", url)
+        representation, from_cache = Representation.cacheable_post(
+            self._db, unicode(url), params, max_age=self.MAX_REPRESENTATION_AGE
+        )
+
+        # Confirm that the representation was successful.
+        if representation.status_code == 403:
+            self._db.delete(representation)
+            raise Exception("Invalid NoveList credentials")
+        if representation.content.startswith('"Missing'):
+            error = representation.content
+            self._db.delete(representation)
+            raise Exception("Invalid NoveList parameters: %s" % error)
+
+        return self.lookup_info_to_metadata(representation.content)
+
+    @classmethod
+    def _scrub_subtitle(self, subtitle):
+        """Removes common NoveList subtitle annoyances"""
+        if subtitle:
+            subtitle = subtitle.replace('[electronic resource]', '')
+            # Then get rid of any leading whitespace or punctuation.
+            subtitle = TitleProcessor.extract_subtitle('', subtitle)
+        return subtitle
+
+    @classmethod
+    def _build_query(cls, params):
+        """Builds a unique and url-encoded query endpoint"""
+
+        for name, value in params.items():
+            params[name] = urllib.quote(value)
+        return (cls.QUERY_ENDPOINT % params).replace(" ", "")
+
+    def lookup_info_to_metadata(self, lookup_info):
+        """Turns a NoveList JSON response into a Metadata object"""
+
+        lookup_info = json.loads(lookup_info)
+        book_info = lookup_info['TitleInfo']
+        if book_info:
+            novelist_identifier = book_info.get('ui')
+        if not book_info or not novelist_identifier:
+            # NoveList didn't know the ISBN. Delete the cache and return None.
+            client_identifier = lookup_info['ClientIdentifier']
+            cached = self._db.query(Representation).\
+                    filter(Representation.url.like(
+                        "%ClientIdentifier="+client_identifier+"%"
+                    ))
```

Comment:
I think it would be better to leave the cached document in place. We can always check again later and use max_age to see if NoveList has new information.